rotkehlchen/api/rest.py

```diff
@@ -4197,3 +4229,41 @@ def get_binance_savings_history(
             query_filter=query_filter,
             value_filter=value_filter,
         )
+
+    def get_ens_avatar(self, ens_name: str, match_header: Optional[str]) -> Response:
+        """
+        Searches for ENS avatar of the given `ens_name`.
+        If found returns the avatar, otherwise returns 404 status code.
```

Comment:
This is a wrong assertion because you return 409 right?
Also, isn't it best you only return either 200, 304 or 404?

I don't think 409 is needed. Thoughts?

Reply:
If there will be no 409, then how to handle errors? Telling the client that there is no avatar when there just a network error occurred will be wrong.

Or if the client requests avatar for an ens name that is not in our database -- we can't just say "no avatar found" because it is a logical error.

Reply:
This error is not user facing, is it?
The error message returned is not used by the frontend. If it's not 200/304 then it falls back to the blocky image.

If this was a json response, yeah that's very useful but it's just blob that's returned here.

Reply:
I would ask @LefterisJP since I think that the same way as we notify users about remote errors when querying transactions or something like that we should also display such errors here.

Reply:
I checked the icons endpoint and it never returns 409. So will probably make the avatars endpoint have the same interface and return 404 in all bad situations.